### DISCUSSION TITLE
Parameters to fixtures

### DISCUSSION THREAD

**Discussion by spuiuk** [pytest-dev/pytest#11775] [Q&A] [answered]
I am fairly new to pytest and am working on a particular code for which I need some advice.

We need to use a setup fixture to setup the test environment which on which we then perform tests. We also need to use the parametrize features to generate the arguments required for the test. These parameters have to be passed to the setup fixture to setup the test environment.

The example below describes the scenario. 
```
import pytest

@pytest.fixture
def myfixture(x, y):
    return x + y

def gen_parameters():
    arr = []
    for i in range(10):
        arr.append((i, i+1))
    return arr

@pytest.mark.parametrize("x,y", gen_parameters())
def test_my(myfixture):
    assert myfixture != 9
```

The result 
```
$ pytest test.py 
============ test session starts =============
platform linux -- Python 3.11.6, pytest-7.2.2, pluggy-1.0.0
rootdir: /tmp, configfile: pytest.ini
collected 10 items                           

test.py::test_my[0-1] PASSED           [ 10%]
test.py::test_my[1-2] PASSED           [ 20%]
test.py::test_my[2-3] PASSED           [ 30%]
test.py::test_my[3-4] PASSED           [ 40%]
test.py::test_my[4-5] FAILED           [ 50%]
test.py::test_my[5-6] PASSED           [ 60%]
test.py::test_my[6-7] PASSED           [ 70%]
test.py::test_my[7-8] PASSED           [ 80%]
test.py::test_my[8-9] PASSED           [ 90%]
test.py::test_my[9-10] PASSED          [100%]
```
In this case, we generate parameters x & y using pytest.mark.parameterize() however the test function itself doesn't use these parameters and instead calls the fixture myfixture() which in turn uses the autogenerated parameters x and y.
This behaviour doesn't seem to be documented anywhere. Is this a valid use or the parametrize functions or is it likely to change in the future. 

We can use the indirect parameter to parametrize and call the fixture that way too. But I was just curious and wanted to know if this is a valid way of using parametrize with fixtures.

**Comment by RonnyPfannschmidt** [comment, marked as answer]
this is a valid way - but its quite confusing - it would be really nice if we had a better way to spell custom pass in parameters for fixtures

a key problem here is that we have a concept of direct/indirect to control whether a parameter should pass to a fixture or override it, but in your case you define a set of new "fixtures" that configure your actual fixture 

currently we want to avoid documenting that capability as we eventually want to come up with a way to spell this nicely

**Comment by spuiuk** [comment]
Thank you.
